tests/e2e/enform.spec.js

```diff
@@ -0,0 +1,293 @@
+        await expect(defaultValField).toHaveValue(formFieldsAttributes[field.name].default_value);
+        await page.locator(`.dialog-${field.id} button[title="Close"]`).click();
+      }
+    }
+
+    const currentUrl = await page.url();
+    enFormId = new URL(currentUrl).searchParams.get('post');
+    console.log(`form ID: ${enFormId}`);
+  });
+
+  test('create Page with EN Form side style', async ({page, context}) => {
+    pageUrl = await createPageWithENForm(page, context, 'side');
+    console.log(`page URL: ${pageUrl}`);
+  });
+
+  test('fill EN Form (side style) on frontend', async ({page, context}) => {
+    await fillENFormAndSubmit(page, context);
+  });
+
+  test('create Page with EN Form full-width-bg style', async ({page, context}) => {
+    pageUrl = await createPageWithENForm(page, context, 'full-width-bg');
+    console.log(`page URL: ${pageUrl}`);
+  });
+
+  test('fill EN Form (full-width-bg style) on frontend', async ({page, context}) => {
+    await fillENFormAndSubmit(page, context);
+  });
+
+  test('disable EN Form block', async ({page, context}) => {
+
+    if (featureIsActiveOnInstance) {
+      console.log('Feature kept active.')
+      return;
+    }
+
+    await page.goto('./wp-admin/');
+    await login(context);
+
+    await page.goto('./wp-admin/admin.php?page=planet4_settings_features');
+    const checkbox = await page.getByRole('checkbox', {name: ' Activate the EN Form block, as well as the "Progress Bar inside EN Form" Counter block style.'});
+    const checked = await checkbox.isChecked();
+    if (checked) {
+      await checkbox.click();
+      await page.locator('input[type="submit"]').click();
+    }
+  });
+});
+
+async function createPageWithENForm(page, context, style) {
+  await page.goto('./wp-admin/');
+  await login(context);
+  await page.goto('./wp-admin/post-new.php?post_type=page');
+
+  // On dev instance, need to close modal so test can continue
+  const closeButton = page.locator('.components-modal__header button');
+  if (await closeButton.isVisible()) {
+    await closeButton.click();
+  }
+
+  await page.locator('.editor-post-title__input').click();
+  await page.locator('h1.editor-post-title').fill('Test page with enform (style)');
+
+  await page.locator('.block-editor-block-list__layout').click();
+  await page.locator('p.is-selected.wp-block-paragraph').fill('/enform');
+  await page.keyboard.press('Enter');
+  await page.waitForSelector('.block.enform-wrap');
+
+  if (style === 'full-width-bg') {
+    await page.getByRole('button', {name: 'Full page width with background'}).click();
+  }
+  const goalSelector = await page.getByRole('combobox', {name: 'Select Goal'});
+  await goalSelector.selectOption('Petition Signup')
+  const formSelector = await page.getByRole('combobox', {name: 'Planet 4 Engaging Networks form'});
+  await formSelector.selectOption(enFormId);
+
+  await page.waitForSelector('#en__field_supporter_emailAddress');
+
+  // Publish page.
```

Comment:
There's now also a [function](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/main/tests/e2e/tools/lib/new-page.js#L22) to publish a page that you could use